mmtrack/datasets/youtube_vis_dataset.py

```diff
@@ -137,3 +145,92 @@ def format_results(self,
         zf.close()
 
         return resfiles, tmp_dir
+
+    def evaluate(self, results, metric=['track_segm'], logger=None):
+        """Evaluation in COCO protocol.
+
+        Args:
+            results (dict): Testing results of the dataset.
+            metric (str | list[str]): Metrics to be evaluated. Options are
+                'track_segm'.
+            logger (logging.Logger | str | None): Logger used for printing
+                related information during evaluation. Default: None.
+
+        Returns:
+            dict[str, float]: COCO style evaluation metric.
+        """
+        if isinstance(metric, list):
+            metrics = metric
+        elif isinstance(metric, str):
+            metrics = [metric]
+        else:
+            raise TypeError('metric must be a list or a str.')
+        allowed_metrics = ['track_segm']
+        for metric in metrics:
+            if metric not in allowed_metrics:
+                raise KeyError(f'metric {metric} is not supported.')
+
+        eval_results = dict()
+        test_results = self.format_results(results, return_json=True)
+        vis_results = self.convert_vis_format()
+        track_segm_results = eval_vis(test_results, vis_results, logger)
+        eval_results.update(track_segm_results)
+
+        return eval_results
+
+    def convert_vis_format(self):
+        """Convert the annotation to the format of YouTube-VIS. The main
+        difference between the two is the format of 'annotation'. Before
+        modification, it is recorded in the unit of images, and after
+        modification, it is recorded in the unit of instances.
+
+        Returns:
+            dict: A dict with 3 keys, ``categories``, ``annotations``
+                and ``videos``.
+            - | ``categories`` (list[dict]): Each dict has 2 keys,
+                ``id`` and ``name``.
+            - | ``videos`` (list[dict]): Each dict has 4 keys of video info,
+                ``id``, ``name``, ``width`` and ``height``.
+            - | ``annotations`` (list[dict]): Each dict has 7 keys of video
+                info, ``category_id``, ``segmentations``, ``bboxes``,
+                ``video_id``, ``areas``, ``id`` and ``iscrowd``.
+        """
+
+        vis_anns = defaultdict(list)
+
+        vis_anns['categories'] = copy.deepcopy(self.coco.dataset['categories'])
+        vis_anns['videos'] = copy.deepcopy(self.coco.dataset['videos'])
+
+        len_videos = dict()  # mapping from video_id to video_length
+        for video_id, video_infos in self.coco.vidToImgs.items():
+            len_videos[video_id] = len(video_infos)
+
+        for video_id, ins_ids in self.coco.vidToInstances.items():
+            cur_video_len = len_videos[video_id]
+            for ins_id in ins_ids:
+                segm = [None] * cur_video_len
```

Comment:
add some description to explain that `segm`、`bbox` and `area` may be None in some frames.